static/app/components/searchSyntax/renderer.tsx

```diff
@@ -36,7 +36,11 @@ function renderResult(result: ParseResult, cursor: number) {
     .map((renderedToken, i) => <Fragment key={i}>{renderedToken}</Fragment>);
 }
 
-function renderToken(token: TokenResult<Token>, cursor: number) {
+function renderToken(token: TokenResult<Token> | string, cursor: number) {
+  if (typeof token === 'string') {
+    return token;
```

Comment:
Why do we need this?

Reply:
because I think we didn't want to have the blueish styles in a free text if it is not invalid...and when it is not invalid, it is of type `string`

https://github.com/getsentry/sentry/blob/1675e2a92757d991fff6ae7b602a2013376e16f8/static/app/components/searchSyntax/renderer.tsx#L208